runtime/src/iree/hal/drivers/cuda/native_executable.c

```diff
@@ -120,12 +120,26 @@ iree_status_t iree_hal_cuda_native_executable_create(
         break;
       }
 
+      int32_t max_shared_mem;
       status = CU_RESULT_TO_STATUS(
           context->syms,
-          cuFuncSetAttribute(function,
-                             CU_FUNC_ATTRIBUTE_MAX_DYNAMIC_SHARED_SIZE_BYTES,
-                             shared_memory_sizes[i]),
-          "cuFuncSetAttribute");
+          cuDeviceGetAttribute(&max_shared_mem,
+                               CU_DEVICE_ATTRIBUTE_MAX_SHARED_MEMORY_PER_BLOCK,
+                               context->cu_device),
+          "cuDeviceGetAttribute");
+      if (shared_memory_sizes[i] > max_shared_mem) {
```

Comment:
this leaks/ignores status - you need to do an `if (!iree_status_is_ok(status)) break;` after each `status =`